### DISCUSSION TITLE
Append media url trait

### DISCUSSION THREAD

**Discussion by royduin** [spatie/laravel-medialibrary#1992] [Ideas] [answered]
For a simple API I'm returning a collection without transforming it with Laravel's build in API Resources or Fractal but I just want to include the media urls without too much code. So I came up with this. Just add the trait to the models where the media urls should be appended automatically.

```php
<?php

namespace App\Traits;

trait AppendFirstMediaUrls
{
    public function attributesToArray()
    {
        $attributes = parent::attributesToArray();
        $this->registerMediaCollections();

        foreach ($this->mediaCollections as $mediaCollection) {
            $attributes['media_'.$mediaCollection->name] = $this->getFirstMediaUrl($mediaCollection->name);
        }

        return $attributes;
    }
}
```

Is this something to have in this package by default? In my current project I only have `singleFile()` media collections but we could check for that and return an array with urls if we've multiple.

Related: https://github.com/spatie/laravel-medialibrary/issues/1135, https://github.com/spatie/laravel-medialibrary/issues/1262 and https://github.com/spatie/laravel-medialibrary/issues/1082

**Comment by freekmurze** [comment, marked as answer]
I'm thinking that it is a bit too opinionated to be in the package. I'd like to keep the package light. I see it as the user responsibility to code up how a media item should be converted to json.

**Reply by royduin** [reply]
Thanks for the answer! 👍
